document/js-api/index.bs

```diff
@@ -687,6 +695,90 @@ which can be simultaneously referenced by multiple {{Instance}} objects. Each
     1. Return undefined.
 </div>
 
+<h3 id="globals">Globals</h3>
+
+<pre class="idl">
+typedef (long or float or double) GlobalValue;
```

Comment:
This isn't allowed as these types aren't distinguishable. (I.e., when accepting a JS value, you can't tell which of these types it should convert to.)

Probably you just want `double`, or maybe `unrestricted double` if you want to allow NaN/+-Infinity.

At that point the typedef isn't so useful.